code/__HELPERS/game.dm

```diff
@@ -233,18 +235,32 @@
 			var/turf/speaker = get_turf(R)
 			if(speaker)
 				for(var/turf/T in hear(R.canhear_range,speaker))
-					speaker_coverage[T] = T
+					if(!speaker_coverage[T])
+						speaker_coverage[T] = R
+						continue
+					if(speaker_coverage[T].canhear_range < R.canhear_range)
+						speaker_coverage[T] = R
```

Comment:
```suggestion
					if(!speaker_coverage[T])
						speaker_coverage[T] = R
						continue
					if(speaker_coverage[T].canhear_range < R.canhear_range)
						speaker_coverage[T] = R
```
Please cast the radios before accessing this. DM doesn't have strongly typed lists, this is basically just tricking spacemandmm not to error on this rather than preventing potential issues.

Reply:
> DM doesn't have strongly typed lists,

*sigh* Really, DM? Why do you make it look like you have a feature you don't?